File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/modules/replaceable/ForegroundReportLifetimeModule.java
package se.gustavkarlsson.skylight.android.dagger.modules.replaceable;

import org.threeten.bp.Duration;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;
import dagger.Reusable;

import static se.gustavkarlsson.skylight.android.dagger.Names.FOREGROUND_REPORT_LIFETIME_NAME;

@Module
public abstract class ForegroundReportLifetimeModule {

	// Published
	@Provides
	@Reusable
	@Named(FOREGROUND_REPORT_LIFETIME_NAME)
	static Duration provideForegroundReportLifetime() {
		return Duration.ofMinutes(15);
	}

}

File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/modules/replaceable/ForegroundReportLifetimeModule.kt
package se.gustavkarlsson.skylight.android.dagger.modules.replaceable

import dagger.Module
import dagger.Provides
import dagger.Reusable
import org.threeten.bp.Duration
import se.gustavkarlsson.skylight.android.dagger.Names.FOREGROUND_REPORT_LIFETIME_NAME
import javax.inject.Named

@Module
class ForegroundReportLifetimeModule {

    // Published
    @Provides
    @Reusable
    @Named(FOREGROUND_REPORT_LIFETIME_NAME)
    fun provideForegroundReportLifetime(): Duration {
        return Duration.ofMinutes(15)
    }

}
